universe/supplycommit/supply_commit_states.go

```diff
@@ -0,0 +1,553 @@
+// version of the supply tree in memory, to stage before going to the next state
+// where we'll create the txn that will be used to commit the new tree in the
+// chain.
+//
+// State transitions:
+//   - CreateTreeEvent -> CommitTxCreateState
+//
+// TODO(roasbeef): have all states cache pending updates once started with tick?
+type CommitTreeCreateState struct {
+}
+
+// stateSealed is a special method that is used to seal the interface.
+func (c *CommitTreeCreateState) stateSealed() {}
+
+// IsTerminal returns true if the target state is a terminal state.
+func (c *CommitTreeCreateState) IsTerminal() bool {
+	return false
+}
+
+// String returns the name of the state.
+func (c *CommitTreeCreateState) String() string {
+	return "CommitTreeCreateState"
+}
+
+// CreateTxEvent is a special event that is used to trigger the creation of the
+// commitment transaction.
+type CreateTxEvent struct {
+}
+
+// eventSealed is a special method that is used to seal the interface.
+func (c *CreateTxEvent) eventSealed() {}
+
+// CommitTxCreateState is the state of the state machine we'll transition to
+// once we've created a new supply tree in memory. In this state, we'll create
+// the actual transaction which spends the existing pre-commitment output(s) and
+// the current commitment output.
+//
+// State transitions:
+//   - CommitTxCreateState -> CommitTxSignState
+type CommitTxCreateState struct {
+	// SupplyTransition holds all the information about the current state
+	// transition, including old/new trees and commitments.
+	SupplyTransition SupplyStateTransition
+}
+
+// TODO(roasbeef): attrs of event vs the state itself?
+
+// stateSealed is a special method that is used to seal the interface.
+func (c *CommitTxCreateState) stateSealed() {}
+
+// IsTerminal returns true if the target state is a terminal state.
+func (c *CommitTxCreateState) IsTerminal() bool {
+	return false
+}
+
+// String returns the name of the state.
+func (c *CommitTxCreateState) String() string {
+	return "CommitTxCreateState"
+}
+
+// SignTxEvent is a special event that is used to trigger the signing of the
+// commitment transaction.
+type SignTxEvent struct {
+	// CommitPkt is the unsigned transaction that will be used to commit to
+	// the new supply tree.
+	CommitPkt *tapsend.FundedPsbt
+
+	// NewSupplyCommit is the new supply commitment that that will be
+	// created by the above commit pkt.
+	NewSupplyCommit RootCommitment
+}
+
+// eventSealed is a special method that is used to seal the interface.
+func (s *SignTxEvent) eventSealed() {}
+
+// CommitTxSignState is the state of the state machine we'll transition to once
+// we've created the tx that commits to the latest supply tree. In this state,
+// we'll sign the transaction and all its inputs, then write our state to disk
+// so we'll be able to recover it on restart.
+//
+// State transitions:
+//   - SignTxEvent -> CommitBroadcastState
+type CommitTxSignState struct {
```

Comment:
I'm not sure if I understand the state transition in the doc comment. Should this be:
```
// State transitions:
//   - SignTxEvent -> CommitTxSignState
//   - CommitTxSignState -> BroadcastEvent
```

Are we describing to and from states in doc `State transitions`?

Reply:
This is meant to indicate that from this state, we can go to the sign state.